File: frontend/lib/models/tache.dart
class Tache {
  final int id;
  final int dossierId;
  final int? responsableId;
  final String titre;
  final String? description;
  final String statut;
  final String priorite;
  final DateTime? dateEcheance;
  final DateTime dateCreation;
  final DateTime? dateTerminaison;
  final bool actif;

  const Tache({
    required this.id,
    required this.dossierId,
    this.responsableId,
    required this.titre,
    this.description,
    required this.statut,
    required this.priorite,
    this.dateEcheance,
    required this.dateCreation,
    this.dateTerminaison,
    required this.actif,
  });

  factory Tache.fromJson(Map<String, dynamic> json) {
    return Tache(
      id: json['id'] as int,
      dossierId: json['dossier_id'] as int,
      responsableId: json['responsable_id'] as int?,
      titre: json['titre'] as String,
      description: json['description'] as String?,
      statut: json['statut'] as String? ?? 'À faire',
      priorite: json['priorite'] as String? ?? 'Normale',
      dateEcheance: json['date_echeance'] != null
          ? DateTime.parse(json['date_echeance'] as String)
          : null,
      dateCreation: DateTime.parse(json['date_creation'] as String),
      dateTerminaison: json['date_terminaison'] != null
          ? DateTime.parse(json['date_terminaison'] as String)
          : null,
      actif: json['actif'] as bool? ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'dossier_id': dossierId,
      'responsable_id': responsableId,
      'titre': titre,
      'description': description,
      'statut': statut,
      'priorite': priorite,
      'date_echeance': dateEcheance?.toIso8601String(),
      'date_creation': dateCreation.toIso8601String(),
      'date_terminaison': dateTerminaison?.toIso8601String(),
      'actif': actif,
    };
  }
}

File: backend/app/models/tache.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Tache(Base):
    __tablename__ = "taches"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)

    dossier_id: Mapped[int] = mapped_column(
        ForeignKey("dossiers.id"),
        nullable=False,
        index=True,
    )

    responsable_id: Mapped[int | None] = mapped_column(
        ForeignKey("users.id"),
        nullable=True,
        index=True,
    )

    titre: Mapped[str] = mapped_column(String(200), nullable=False)

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True,
    )

    statut: Mapped[str] = mapped_column(
        String(50),
        nullable=False,
        default="À faire",
    )

    priorite: Mapped[str] = mapped_column(
        String(50),
        nullable=False,
        default="Normale",
    )

    date_echeance: Mapped[datetime | None] = mapped_column(
        DateTime,
        nullable=True,
    )

    date_creation: Mapped[datetime] = mapped_column(
        DateTime,
        nullable=False,
        default=datetime.utcnow,
    )

    date_terminaison: Mapped[datetime | None] = mapped_column(
        DateTime,
        nullable=True,
    )

    actif: Mapped[bool] = mapped_column(
        Boolean,
        nullable=False,
        default=True,
    )

    dossier = relationship(
        "Dossier",
        back_populates="taches",
    )

    responsable = relationship(
        "User",
        back_populates="taches",
    )
